### PR TITLE
Update Fuzzer documentation and Configuration

### PR DESCRIPTION
**Pull request by Logix64**
This references #4061. With these changes both ASL and libFuzzer build on my device. Extending on that, I would like to continue to update the fuzzing source code, but I don't know how exactly (especially the RNG part). 

Any help would be appreciated.

### CODE REVIEW

File: src/fuzzer/fuzzers.h [hunk 1]
```diff
@@ -33,16 +33,16 @@ extern "C" int LLVMFuzzerInitialize(int*, char***) {
...
 }
 
 // Some helpers for the fuzzer jigs
 
-inline std::shared_ptr<Botan::RandomNumberGenerator> fuzzer_rng_as_shared() {
-   static std::shared_ptr<Botan::ChaCha_RNG> rng =
-      std::make_shared<Botan::ChaCha_RNG>(Botan::secure_vector<uint8_t>(32));
+inline std::shared_ptr<Botan::System_RNG> fuzzer_rng_as_shared() {
+   static std::shared_ptr<Botan::System_RNG> rng =
+      std::make_shared<Botan::System_RNG>();
```
Comment (randombit) [inline src/fuzzer/fuzzers.h:45]: This definitely doesn't work - the non-determinism of the RNG will confuse the fuzzer. Lets revert this bit and then take up the RNG piece in a future PR. I'll add more context in the original ticket

**Review by randombit — COMMENTED** [review]
Hi @Logix64 thanks so much for taking this on! This is definitely a nice improvement especially the span bit. There are a couple trivial warnings from `clang-tidy` and some formatting things (there is a diff in the `format` CI job) that need to be addressed. For the RNG part I'll give more context in the original ticket.

File: src/fuzzer/fuzzers.h (hunk 1 above)
Comment (Logix64) [reply src/fuzzer/fuzzers.h:45]: Okay, thank you. I changed it in my recent commit.

**Comment by randombit** [comment]
Thanks @Logix64 this is a nice cleanup. Can you rebase then squash the last 3 commits? Other than that this looks to me ready to merge.

**Comment by randombit** [comment]
Agh I'm sorry @Logix64 I delayed this to merge until after 3.5.0 but now some merge conflicts have come in. Can you rebase?

**Comment by Logix64** [comment]
> Agh I'm sorry @Logix64 I delayed this to merge until after 3.5.0 but now some merge conflicts have come in. Can you rebase?

Yeah no problem, I got it fixed now. The second force-push was for a new case with the old signature (mp_redc_crandall). Is the RNG issue already fixed? Otherwise I would continue working on that.

**Review by randombit — APPROVED** [review]
Sorry for the delay and thanks again for working on this!

**Pull request merged**